fieldpath/fieldpath-gen/template.go

```diff
@@ -1121,10 +1121,11 @@ func (this *{{.CName}}SinglePath) UnmarshalFirst(buf []byte) ({{.ReturnType}}, e
 				endOfs[position] = offset + int(length)
 			}
 		} else {
-			offset, err = skip(buf, offset, wireType)
+			newOffset, err := skip(buf, offset, wireType)
```

Comment:
Isn't this also weird.
I am allowed to the offset, below, but not at the actual place I want to set it?